commits.go

```diff
@@ -56,49 +51,73 @@ func (commitsTable) Schema() sql.Schema {
 	return CommitsSchema
 }
 
-func (r *commitsTable) TransformUp(f sql.TransformNodeFunc) (sql.Node, error) {
-	return f(r)
+func (r *commitsTable) WithFilters(filters []sql.Expression) sql.Table {
+	nt := *r
+	nt.filters = filters
+	return &nt
 }
 
-func (r *commitsTable) TransformExpressionsUp(f sql.TransformExprFunc) (sql.Node, error) {
-	return r, nil
-}
-
-func (r commitsTable) RowIter(ctx *sql.Context) (sql.RowIter, error) {
-	span, ctx := ctx.Span("gitbase.CommitsTable")
-	iter := new(commitIter)
-
-	repoIter, err := NewRowRepoIter(ctx, iter)
-	if err != nil {
-		span.Finish()
-		return nil, err
-	}
-
-	return sql.NewSpanIter(span, repoIter), nil
+func (r *commitsTable) WithProjection(colNames []string) sql.Table {
+	return r
```

Comment:
doesn't accept this table pushdowns of projections?

Reply:
same answer